src/instance.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{
+    errors::to_py_err, exports::Exports, import_object::ImportObject, module::Module,
+    wasmer_inner::wasmer,
+};
+use pyo3::{exceptions::RuntimeError, prelude::*};
+
+#[pyclass(unsendable)]
+#[text_signature = "(module)"]
+pub struct Instance {
+    inner: wasmer::Instance,
+    exports: Py<Exports>,
+}
+
+pub enum InstanceError {
+    InstantiationError(wasmer::InstantiationError),
+    PyErr(PyErr),
+}
+
+impl Instance {
+    pub fn raw_new(
+        py: Python,
+        module: &Module,
+        import_object: Option<&ImportObject>,
+    ) -> Result<Self, InstanceError> {
+        let instance = match import_object {
+            Some(import_object) => wasmer::Instance::new(module.inner(), import_object.inner()),
+            None => wasmer::Instance::new(module.inner(), &wasmer::imports! {}),
+        };
```

Comment:
Hmm, I didn't originally see the `inner()` call so I like this less than I originally did, but you can also probably do

```Rust
let import_object = import_object.map(|io| io.inner()).unwrap_or_else(|| wasmer::imports! {});
wasmer::Instance::new(module.inner(), import_object);
```

Which is a little cleaner, there may be lifetime issues with this though depending on the signature of `inner`

Reply:
The problem is that we have to pass `ImportObject` by reference, and in case where we create an empty `ImportObject`, there is a lifetime issue. That's the only way I found to fix that.

Reply:
If `imports!{}` doesn't heap allocate, then you can solve it by adding `let empty_imports = imports!{}` and then referencing that internally, but yeah

Reply:
Not a big deal, I think it's fine as-is